src/iothubclient/iothub_client.go

```diff
@@ -42,6 +42,19 @@ func SetStaticDeviceInfo(c *iotdevice.Client, d DeviceInfo) error {
 	return nil
 }
 
+// Wait forever for IotHub Connection
+func waitForIotHubConnection(c *iotdevice.Client) error {
+	for {
+		err := c.Connect(context.Background())
+
+		if err == nil {
```

Comment:
You can remove the return value (error), as only the positive way is returned (== nil). 
Remove the empty line 49 to indicate the err (line 48) belongs to the error handling (line 50).